src/ServiceControl.Monitoring/Http/Diagrams/TimingsAggregator.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ServiceControl.Monitoring.Timings
+{
+    class TimingsAggregator
+    {
+        public static List<MonitoredEndpoint> AggregateIntoLogicalEndpoints(ProcessingTimeStore processingTimeStore, CriticalTimeStore criticalTimeStore)
```

Comment:
I wonder if this code couldn't look a bit better by changing the return type to `IEnumerable` and then `yield return`ing the MonitoredEndpoint instance instead of adding to a full list